uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/my-leave/components/my-leave-calendar.js

```diff
@@ -9,21 +11,352 @@ define([
+      return !!publicHolidays[new Date(date).getTime()];
+    };
+
+    /**
+     * Labels the given period according to whether it's current or not
+     *
+     * @param  {AbsencePeriodInstance} period
+     * @return {string}
+     */
+    vm.labelPeriod = function (period) {
+      return period.current ? 'Current Period (' + period.title + ')' : period.title;
+    };
+
+    /**
+     * Returns day name of the sent date(Monday, Tuesday etc.)
+     *
+     * @param  {string} date
+     * @return {string}
+     */
+    vm.getDayName = function (date) {
+      return new Date(date).toString().substr(0,3);
+    };
+
+    /**
+     * Returns the calendar information for a specific month
+     *
+     * @param  {int} month
+     * @return {array}
+     */
+    vm.getMonthData = function (month) {
+      if (vm.calendar.days) {
+        var iterator,
+          dates = Object.keys(vm.calendar.days),
+          date,
+          length = dates.length,
+          datesForTheMonth = [];
+
+        for (iterator = 0; iterator < length; iterator++) {
+          date = new Date(parseInt(dates[iterator]));
+          if (date.getMonth() === month) {
+            datesForTheMonth.push(vm.calendar.days[dates[iterator]]);
+          }
+        }
+
+        return datesForTheMonth;
+      }
+    };
+
+    /**
+     * Refresh all leave request and calendar data
+     */
+    vm.refresh = function () {
+      vm.loading = true;
+      loadLeaveRequest()
+        .then(function () {
+          vm.loading = false;
+        });
+    };
+
+    (function init() {
+      vm.loading = true;
+      $q.all([
+        loadAbsencePeriods(),
+        loadAbsenceTypes(),
+        loadPublicHolidays(),
+        loadStatuses(),
+        loadDayTypes()
+      ]).then(function () {
+          vm.legendCollapsed = false;
+          return loadLeaveRequest();
+        })
+        .then(function () {
+          vm.loading = false;
+        });
+    })();
+
+    /**
+     * Returns the leave request which is in range of the sent date
+     *
+     * @param  {string} date
+     * @return {object}
+     */
+    function getLeaveRequest(date) {
+      var iterator,
+        length = leaveRequests.length,
+        dates;
+
+      for (iterator = 0; iterator < length; iterator++) {
+        dates = _.find(leaveRequests[iterator].dates, function (leaveRequestDate) {
+          return moment(leaveRequestDate.date).isSame(date);
+        });
+
+        if (dates) {
+          return leaveRequests[iterator];
+        }
+      }
+    }
+
+    /**
+     * Returns the styles for a specific leaveRequest
+     * which will be used in the view for each date
+     *
+     * @param  {object} leaveRequest
+     * @return {object}
+     */
+    function getStyles(leaveRequest) {
+      var status = leaveRequestStatuses[leaveRequest.status_id],
+        absenceType;
+
+      if (status.name === 'waiting_approval'
+        || status.name === 'approved'
+        || status.name === 'admin_approved') {
```

Comment:
or `else`? It seems you have various functions where you do not specify what should happen if a certain `if` is not satisfied